candidate/models.py

```diff
@@ -0,0 +1,93 @@
+from __future__ import unicode_literals
+from django.db import models
+from django.utils.encoding import python_2_unicode_compatible
+
+from ballot_measure.models import BallotMeasureChoice
+
+
+class SocialMediaModel(models.Model):
+    class Meta:
+        abstract = True
+
+    photo_url = models.ImageField(blank=True)
+    website_url = models.URLField(
+        help_text='URL for the official website.', blank=True)
+    facebook_url = models.URLField(
+        help_text='URL for the Facebook page.', blank=True)
+    twitter_url = models.URLField(
+        help_text='URL for the Twitter page.', blank=True)
+
+
+@python_2_unicode_compatible
+class Party(SocialMediaModel):
+    """
+    """
+    name = models.CharField(
+        max_length=255, help_text='The party name.')
+
+    def __str__(self):
+        return "The %s party" % self.name
+
+
+@python_2_unicode_compatible
+class Person(SocialMediaModel):
+    """
+    """
+    first_name = models.CharField(
+        max_length=255, help_text='The candidate\'s first name.')
+    middle_name = models.CharField(
+        max_length=255, blank=True, null=True,
+        help_text='The candidate\'s middle name.')
+    last_name = models.CharField(
+        max_length=255, blank=True, null=True,
+        help_text='The candidate\'s last name.')
+    party = models.ForeignKey('Party', blank=True, null=True)
+
+    def __str__(self):
+        return "%s, %s%s" % (
+            self.last_name, self.first_name,
+            '' if self.middle_name is None else ' %s.' % self.middle_name[0])
+
+
+@python_2_unicode_compatible
+class Office(models.Model):
+    """
+    """
+    name = models.CharField(
+        max_length=255, help_text='The office name.')
+    description = models.CharField(
+        max_length=1024, help_text='The office description.')
+    locality = models.ForeignKey('locality.Locality')
+
+    def __str__(self):
+        return "Office for %s in %s" % (
+            self.name, self.locality.dereference())
+
+
+@python_2_unicode_compatible
+class Election(models.Model):
+    """
+    """
+    office = models.ForeignKey('Office')
+    ballot_measure = models.ForeignKey(
+        'ballot_measure.BallotMeasure')
+
+    def locality(self):
+        return self.ballot_measure.ballot.locality
+
+    def __str__(self):
+        return "Election of %s in %s" % (
+            self.office.name, self.locality().dereference())
+
+
+class Candidate(BallotMeasureChoice, SocialMediaModel):
```

Comment:
A lot of classes inherit from `SocialMediaModel`, which, I suppose, is taking advantage of Python's multiple inheritance feature.

Is there a modular way to mix in methods to classes in python? I only ask because I've recently jumped on the [Composition over Inheritance](https://en.wikipedia.org/wiki/Composition_over_inheritance) bandwagon.